manifests/init.pp

```diff
@@ -1,179 +1,270 @@
-    if empty($nmcli_connection_name) {
-      fail('Cannot modify DNS servers via nmcli unless a connection name is specified. Please ensure resolv::nmcli_connection_name is set to a valid network connection name')
-    }
+      # Make sure the user has specified a connection to modify
+      if empty($nmcli_connection_name) {
+        fail('Cannot modify DNS servers via nmcli unless a connection name is specified. Please ensure resolv::nmcli_connection_name is set to a valid network connection name')
+      }
 
-    $device_names = dig($facts, 'simplib__networkmanager', 'connection')
+      $device_names = dig($facts, 'simplib__networkmanager', 'connection')
 
-    # Check to ensure Network Manager is active
-    if ! $device_names {
-      fail('There are no connections being managed by Network Manager which are able to be modified')
-    }
+      # Check to ensure Network Manager is active
+      if ! $device_names {
+        fail('There are no connections being managed by Network Manager which are able to be modified')
+      }
 
-    # Check to ensure the user-specified connection is managed by Network Manager
-    if ! $device_names.any |$name, $properties| {
-      $properties['name'] == $nmcli_connection_name
-    } {
-      fail("The specified connection: \"${nmcli_connection_name}\" is not managed by Network Manager and cannot be modified")
-    }
+      # Check to ensure the user-specified connection is managed by Network Manager
+      if ! $device_names.any |$name, $properties| {
+        $properties['name'] == $nmcli_connection_name
+      } {
+        fail("The specified connection: \"${nmcli_connection_name}\" is not managed by Network Manager and cannot be modified")
+      }
 
-    $device_names.each |String $device_name, Hash $conn_properties| {
-      if $conn_properties['name'] == $nmcli_connection_name {
-        $_flattened_name_servers = $servers.join(' ')
+      $device_names.each |String $device_name, Hash $conn_properties| {
+        if $conn_properties['name'] == $nmcli_connection_name {
+          $_flattened_name_servers = $servers.join(' ')
 
-        # Add the specified nameservers unless they are already configured for the given device
-        exec { 'Add DNS servers via nmcli':
-          command => "nmcli connection modify \"${nmcli_connection_name}\" ipv4.ignore-auto-dns ${nmcli_ignore_auto_dns} ipv4.dns \"${_flattened_name_servers}\"",
-          unless  => "[ \"\$( nmcli -f ipv4.dns connection show \"${nmcli_connection_name}\" | awk '{print \$2}' | tr '\\n' ' ' | tr ',' ' ' )\" == \"${_flattened_name_servers} \" ]",
-          path    => '/bin:/usr/bin',
-        }
+          # Add the specified nameservers unless they are already configured for the given device
+          exec { 'Add DNS servers via nmcli':
+            command => "nmcli connection modify \"${nmcli_connection_name}\" ipv4.ignore-auto-dns ${nmcli_ignore_auto_dns} ipv4.dns \"${_flattened_name_servers}\"",
```

Comment:
Do you also want to add the search domains to the connection?

Reply:
Yeah, it looks like we might want to pass all of the various options into nmcli. It's a shame that it's so clunky.